### DISCUSSION TITLE
502 Bad Gateway sonarr/radarr reverse proxy

### DISCUSSION THREAD

**Discussion by mluker** [NginxProxyManager/nginx-proxy-manager#3505] [Q&A]
My set up:
* I am running docker compose with radarr, sonarr, and nginx proxy manager
* I am using PiHole for DNS, all sites resolve to the correct IP address
* I am running all internal, no outside access
* No changes to the compose setting (posted below)

When I access any of the sites (sonarr/radarr) via IP:port they load quick and everything works. When I use the fqdn (routed through proxy manager) https://sonarr.mydomain.host the site loads very slow and usually throws a bad gateway. The bad gateway comes in the way of the entire request or individual requests. I am forcing SSL, enabled websockets, and enabled HTTP/2. Any ideas on what might cause this sporadic and slow resolution from the proxy?

  npm:
    container_name: npm
    image: 'jc21/nginx-proxy-manager:latest'
    restart: always
    ports:
      # These ports are in format <host-port>:<container-port>
      - '80:80' # Public HTTP Port
      - '443:443' # Public HTTPS Port
      - '81:81' # Admin Web Port
    volumes:
      - ./npm/data:/data
      - ./letsencrypt:/etc/letsencrypt
    environment:
      - PUID=0
      - PGID=0
      - UMASK=${UMASK:?err}
      - TZ=${TIMEZONE:?err}

**Comment by Snoresville** [comment]
Is ur sonarr on the same docker network as nginx proxy manager? Actually, the two services need to be on the same network so that nginx proxy manager can see the sonarr and reverse proxy into it
